README.md
<b>LeviticusSeed</b>
====================
<br>
Linux .Bash Miner-Run Management Utility
<br>
Currently In Pre-Beta & Was Coded For Personal Use
<br>
Public Version Coming Soon
<br>
If you know some .Bash code I'm sure you will find this useful; If not please be patient as I finish the beta
<br>
<br>
  <p align="center">
    <img src="http://i.imgur.com/KaFxyJf.png">
  </p>
============
<b>Usage:</b>
=============
<br>
  <b><i>Option-A (Copy & Paste):</i></b><br>
    &nbsp;1. Enter GitHub Source Of LS-0.0.2.sh<br>
    &nbsp;2. Select All & Copy<br>
    &nbsp;3. In Terminal Of Linux -or RaspberryPi<br>
      &nbsp; &nbsp; &nbsp;A. <i>cd</i><br>
      &nbsp; &nbsp; &nbsp;B. <i>nano LS.sh</i><br>
      &nbsp; &nbsp; &nbsp;C. (Paste Entire Source)<br>
      &nbsp; &nbsp; &nbsp;D. Save: Ctrl + O<br>
      &nbsp; &nbsp; &nbsp;E. Exit: Ctrl + X<br>
      &nbsp; &nbsp; &nbsp;F. Premissions: <i>chmod +x LS.sh</i><br>
      &nbsp; &nbsp; &nbsp;G. Run: <i>./LS.sh</i> or <i>sudo ./LS.sh</i><br>
  
 <b><i>Option-B (Git Clone):</i></b><br>
     &nbsp;1. <i>git clone https://github.com/Proteu5/LeviticusSeed.git</i><br>
     &nbsp;2. <i>cd LeviticusSeed</i><br>
     &nbsp;3. Premissions: </i>chmod +x LS-0.0.2.sh</i><br>
     &nbsp;4. <i>./LS-0.0.2.sh</i><br>
     &nbsp;5. Create *.sh Miner Execution Files<br>
     &nbsp; &nbsp; &nbsp; A. Don't Forget To Set Premission To Executable<br>
     &nbsp;6. <i>./LS-0.0.2.sh</i><br>
  
  <b><i>Option-C (~/.bashrc Terminal-AutoRun):</i></b><br>
    <i>&nbsp;X. UnStable But Possible</i><br>
     &nbsp;1. <i>sudo nano ~/.bashrc</i><br>
     &nbsp;2. @Bottom: <i>/root/LeviticusSeed/.LS-0.0.2.sh</i><br>
     &nbsp;3. Save & Exit: Ctrl + O & Ctrl + X<br>
     &nbsp;4. <i>Sudo Screen</i><br>
     &nbsp;5. LS-0.0.2.sh Should Run Each Time Terminal Starts: Similar To 'screenfetch' or 'archey'<br>
     &nbsp;6. I will be updating for further stability for this implementation<br>
    
    
    


ChangeLog
...:::Future Beta Release:::...

LeviticusSeed-0.0.3 (Coming Soon...)

  1. Control Panal Loop To Avoid Relaunching './LS.sh'
  2. Ease Of Use & Stability For Use In ~/.bashrc Terminal: AutoRun
  3. Built-In *.Sh File Creator/Editor To Generate Script Files In Root or Local LeviticusSeed Directory


LeviticusSeed-0.0.2
  - Added Control Panal [l]ist [g]uide [h]elp [i]nfo
  - Added Comments
  - Semi-Stability
  - Error Loop
  - Known ./sh Launcher Bugs (Fixing For Beta)
  
LeviticusSeed-0.0.1
  - Added ASCII Art
  - Added Date & Unix Time Stamp
  - Added *.sh File Launcher With $RESP


LS-0.0.2.sh
now=$(date)
nowunix=$(date +%s)

# Color Codes #
red='\e[0;31m'
NC='\e[0m' # No Color
Black='\e[0;30m'
Blue='\e[0;34m'
LBlue='\e[1;34m'
Green='\e[0;32m'
LGreen='\e[1;32m'
Cyan='\e[0;36m'
LCyan='\e[1;36m'
Red='\e[0;31m'
LRed='\e[1;31m'
Yellow='\e[1;33m'
Purple='\e[0;35m'
LPurple='\e[1;35m'
Brown='\e[0;33m'
LightGray='\e[0;37m'
DGray='\e[1;30m'
White='\e[1;37m'

#      . . . : : : LS Program : : : . . .     #
#< - >< - >< - >< - >< - >< - >< - >< - >< - >#
# Run./LS-0.0.2.sh                            #
# Type: zus1                                  #
# zus1.sh Script Will Run                     #
###############################################

sleep 1
clear
echo --------------------------------------------------------------------------------
echo "--------------------------------------------------------------------------------";
echo "---                                           __                             ---";
echo "---                      /  _    '_/'_    _  (  _ _ _/                       ---";
echo "---                     (__(- \// //( (/_)  __)(-(-(/                        ---";
echo "---                        Version - 0.0.2 : Proteu5                         ---";
echo "---                                                                          ---";
echo "--- "$now"                                  "$nowunix" ---"; 
echo "--------------------------------------------------------------------------------";
echo "----------------------Leviticus Seed Terminal {42} Sub-Net----------------------";
echo
echo -e "${LightGrey}|${NC}" "${Cyan}Welcome To L.S. SubNet Command Terminal${NC}" "${LightGrey}|${NC}   " ["${Cyan}L${NC}"]ist ["${Cyan}G${NC}"]uide ["${Cyan}H${NC}"]elp ["${Cyan}I${NC}"]nfo "   ${LightGrey}|${NC}"
echo
read -p "Leviticus Is Waiting... " RESP

# Edit Below For Customized [l]ist Of Of *.Sh Miner Files #
if [ "$RESP" = "l" ]; then
  echo "Enumerate Miner *.sh Files Here"
  echo "ZeusMiner-001: Blizzard [zus1]"
  echo "ZeusMiner-002: ZusX3 [zus2]"
  echo "RockMiner-001: R-Box [rkm1]"
  echo "GridSeedBlade-001: Blazer80 [blz1]"
  echo "This Is Your List Of Units And Their HotKeys"
  
# Built In [g]uide For Additional Assistance #
elif [ "$RESP" = "g" ]; then
  echo "We Are Currently In Pre-Beta: V-0.0.2"
  echo "The Program Does Not Loop"
  echo "After Running Any Command You Will Need To ReLaunch LS"
  echo "------------------------------------------------------"
  echo "File-Structure Setup And Compatability"
  echo "  1. For Each Desired Unit You Will Need To Code A *.sh File In Your Root-Dir"
  echo "     a. Terminal: nano zus1.sh"
  echo "     b. Type: echo Priming Device Zus-001"
  echo "     c. NxtLine: sudo -s"
  echo "     d. NxtLine: cd"
  echo "     e. NxtLine: cd /root/bfg-zeus/bfgminer"
  echo "     f. NxtLine: ./bfgminer -c /root/zeus-bfg/bfgminer/zus1.conf"
  echo "     g. Save: [CTRL] + [O]"
  echo "     h. Exit: [CTRL] + [X]"
  echo "     i. SetExecutable: chmod +x zus1.sh"
  echo "     j. Test: sudo ./zus1.sh"
  echo "  2. ReCode The LeviticusSource To Your FileSystem Structure"
  echo "- - - - - - - - - - - - - - - - - - - - - - - - - - -"
  
# [h]elp Display For Useful Info & Tips #
elif [ "$RESP" = "h" ]; then
  echo ". . . : : : HelpScreen : : : . . ."
  echo "Required Programs:"
  echo "  1. SSH: sudo apt-get install ssh"
  echo "  2. Screen: sudo apt-get install screen"
  echo "  3. More To Come"
  
# [i]nfo Section On Dev. #
elif [ "$RESP" = "i" ]; then
  echo "Coded By: Proteu5"

# # # EDIT Below As-Per Your Specific FileSystem Structure            # # #
# # # elif = else if conditional command                              # # #
# # # [ "$RESP" = "zus1" ]; then = User Input Field & 'Then' Command  # # #
# # # Change "zus1" To Your Own Desired Command: 001 or Miner01 etc.  # # #
# # # echo "Running..." = A Breif Display Of Text, Good For Debugging # # #
# # # (sudo ./zus1.sh) = The *.Sh File You Wish To Run                # # #
# # # If LS-0.0.2.sh Is in /root/ and so are your *.sh's follow this  # # #
# # # Else, If your *.sh's are elsewhere: "cd /root/bfgminer/" needed # # #
elif [ "$RESP" = "zus1" ]; then
   echo "Running ZeusMiner-001"
   (sudo ./zus1.sh)
elif [ "$RESP" = "zus2" ]; then
   echo "Running ZeusMiner-002"
   (sudo ./zus2.sh)
elif [ "$RESP" = "rkm1" ]; then
   echo "Running RockMiner-001"
   (sudo ./rkm1.sh)
elif [ "$RESP" = "blz1" ]; then
   echo "Running GridSeedBlade-001:Blazr80"
   (sudo ./blz1.sh)
elif [ "$RESP" = "BTC0" ]; then
   echo Running BTC-0 QuadMiners 
   `(gnome-terminal -e sudo /home/orion/Desktop/LeviticusSeed/ exec ./BitParkingBTCNMCDEVOrionPrime.sh)`
else
  echo "ERROR: We need more bash programming"
  ./LS-0.0.2.sh #Looping Program On Error
fi


LS_Maker.sh
  read -p "Enter Your Handle: " myid
  read -p "Enter Device Identifier: " devid
  read -p "Enter Desiered Coin To Mine: " coinid
  read -p "Enter: [b]fgminer or [c]gminer " mineid
  read -p "Enter Path Of BFG/CGMiner " minerid
  read -p "Enter Pool: " poolid
  read -p "Enter Worker.Name: " workerid
  read -p "Enter Worker PassWord: " passid
  read -p "Enter Additional Parameters: " hashid
  
  echo "echo Priming System For $coinid" > /tmp/$devid.sh
  echo "sleep 1" >> /tmp/$devid.sh
  echo "echo Welcome $myid" >> /tmp/$devid.sh
  if [ "$mineid" = "b" ]; then
      echo "cd /root/bfgminer" >> /tmp/$devid.sh
      echo "sudo ./bfgminer -o $poolid -u $workerid -p $passid $hashid" >> /tmp/$devid.sh
      chmod +x /tmp/$devid.sh
  elif  [ "$mineid" = "c" ]; then
     echo "cd /root/cgminer" >> /tmp/$devid.sh
     echo "sudo ./cgminer -o $poolid -u $workerid -p $passid $hashid" >> /tmp/$devid.sh
     chmod +x /tmp/$devid.sh
  else 
     echo "Miner Parth Error: Please Nano/Vi *.Sh File Manually"
fi


LS-0.0.3-PreRelease.sh
now=$(date)
nowunix=$(date +%s)

# Color Codes #
red='\e[0;31m'
NC='\e[0m' # No Color
Black='\e[0;30m'
Blue='\e[0;34m'
LBlue='\e[1;34m'
Green='\e[0;32m'
LGreen='\e[1;32m'
Cyan='\e[0;36m'
LCyan='\e[1;36m'
Red='\e[0;31m'
LRed='\e[1;31m'
Yellow='\e[1;33m'
Purple='\e[0;35m'
LPurple='\e[1;35m'
Brown='\e[0;33m'
LightGray='\e[0;37m'
DGray='\e[1;30m'
White='\e[1;37m'

#      . . . : : : LS Program : : : . . .     #
#< - >< - >< - >< - >< - >< - >< - >< - >< - >#
# Run./LS-0.0.2.sh                            #
# Type: zus1                                  #
# zus1.sh Script Will Run                     #
###############################################

sleep 1
clear
echo --------------------------------------------------------------------------------
echo "--------------------------------------------------------------------------------";
echo "---                                           __                             ---";
echo "---                      /  _    '_/'_    _  (  _ _ _/                       ---";
echo "---                     (__(- \// //( (/_)  __)(-(-(/                        ---";
echo "---                        Version - 0.0.2 : Proteu5                         ---";
echo "---                                                                          ---";
echo "--- "$now"                                  "$nowunix" ---"; 
echo "--------------------------------------------------------------------------------";
echo "----------------------Leviticus Seed Terminal {42} Sub-Net----------------------";
echo
echo -e "${LightGrey}|${NC}" "${Cyan}Welcome To L.S. SubNet Command Terminal${NC}" "${LightGrey}|${NC}   " ["${Cyan}L${NC}"]ist ["${Cyan}G${NC}"]uide ["${Cyan}H${NC}"]elp ["${Cyan}I${NC}"]nfo "   ${LightGrey}|${NC}"
echo
read -p "Leviticus Is Waiting... " RESP

# Edit Below For Customized [l]ist Of Of *.Sh Miner Files #
if [ "$RESP" = "l" ]; then
  echo "Enumerate Miner *.sh Files Here"
  echo "ZeusMiner-001: Blizzard [zus1]"
  echo "ZeusMiner-002: ZusX3 [zus2]"
  echo "RockMiner-001: R-Box [rkm1]"
  echo "GridSeedBlade-001: Blazer80 [blz1]"
  echo "Example-01: ExampleSH [exp]"
  echo "This Is Your List Of Units And Their HotKeys"
  
# Built In [g]uide For Additional Assistance #
elif [ "$RESP" = "g" ]; then
  echo "We Are Currently In Pre-Beta: V-0.0.2"
  echo "The Program Does Not Loop"
  echo "After Running Any Command You Will Need To ReLaunch LS"
  echo "------------------------------------------------------"
  echo "File-Structure Setup And Compatability"
  echo "  1. For Each Desired Unit You Will Need To Code A *.sh File In Your Root-Dir"
  echo "     a. Terminal: nano zus1.sh"
  echo "     b. Type: echo Priming Device Zus-001"
  echo "     c. NxtLine: sudo -s"
  echo "     d. NxtLine: cd"
  echo "     e. NxtLine: cd /root/bfg-zeus/bfgminer"
  echo "     f. NxtLine: ./bfgminer -c /root/zeus-bfg/bfgminer/zus1.conf"
  echo "     g. Save: [CTRL] + [O]"
  echo "     h. Exit: [CTRL] + [X]"
  echo "     i. SetExecutable: chmod +x zus1.sh"
  echo "     j. Test: sudo ./zus1.sh"
  echo "  2. ReCode The LeviticusSource To Your FileSystem Structure"
  echo "- - - - - - - - - - - - - - - - - - - - - - - - - - -"
  
# [h]elp Display For Useful Info & Tips #
elif [ "$RESP" = "h" ]; then
  echo ". . . : : : HelpScreen : : : . . ."
  echo "Required Programs:"
  echo "  1. SSH: sudo apt-get install ssh"
  echo "  2. Screen: sudo apt-get install screen"
  echo "  3. More To Come"
  
# [i]nfo Section On Dev. #
elif [ "$RESP" = "i" ]; then
  echo "Coded By: Proteu5"

elif [ "$RESP" = "p" ]; then
  ('/home/orion/LeviticusSeed-master/LS_Maker.sh')

##### EDIT Below As-Per Your Specific FileSystem Structure            #####
# # # elif = else if conditional command                              # # #
# # # [ "$RESP" = "zus1" ]; then = User Input Field & 'Then' Command  # # #
# # # Change "zus1" To Your Own Desired Command: 001 or Miner01 etc.  # # #
# # # echo "Running..." = A Breif Display Of Text, Good For Debugging # # #
# # # (sudo ./zus1.sh) = The *.Sh File You Wish To Run                # # #
# # # If LS-0.0.2.sh Is in /root/ and so are your *.sh's follow this  # # #
##### Else, If your *.sh's are elsewhere: "cd /root/bfgminer/" needed #####
elif [ "$RESP" = "zus1" ]; then
   echo "Running ZeusMiner-001"
   (sudo ./zus1.sh)
elif [ "$RESP" = "exp" ]; then
   echo "Running Example01"
   (sudo '/home/orion/LeviticusSeed-master/example01.sh')
elif [ "$RESP" = "zus2" ]; then
   echo "Running ZeusMiner-002"
   (sudo ./zus2.sh)
elif [ "$RESP" = "rkm1" ]; then
   echo "Running RockMiner-001"
   (sudo ./rkm1.sh)
elif [ "$RESP" = "blz1" ]; then
   echo "Running GridSeedBlade-001:Blazr80"
   (sudo ./blz1.sh)
elif [ "$RESP" = "BTC0" ]; then
   echo Running BTC-0 QuadMiners 
   `(gnome-terminal -e sudo /home/orion/Desktop/LeviticusSeed/ exec ./BitParkingBTCNMCDEVOrionPrime.sh)`
else
  echo "ERROR: We need more bash programming"
  '/home/orion/LeviticusSeed-master/LS-0.0.2.sh' #Looping Program On Error
fi
